diff --git a/ui/build/soong.go b/ui/build/soong.go
index c9b303c..bedfbc1 100644
--- a/ui/build/soong.go
+++ b/ui/build/soong.go
@@ -226,6 +226,12 @@ func (pb PrimaryBuilderFactory) primaryBuilderInvocation(config Config) bootstra
 
 	commonArgs = append(commonArgs, "-l", filepath.Join(pb.config.FileListDir(), "Android.bp.list"))
 	invocationEnv := make(map[string]string)
+	if goMemLimit := os.Getenv("SOONG_GOMEMLIMIT"); goMemLimit != "" {
+		invocationEnv["GOMEMLIMIT"] = goMemLimit
+	}
+	if goGC := os.Getenv("SOONG_GOGC"); goGC != "" {
+		invocationEnv["GOGC"] = goGC
+	}
 	if pb.debugPort != "" {
 		//debug mode
 		commonArgs = append(commonArgs, "--delve_listen", pb.debugPort,
