diff --git a/bootstrap/bootstrap.go b/bootstrap/bootstrap.go
index 72dc538..5bcfb9d 100644
--- a/bootstrap/bootstrap.go
+++ b/bootstrap/bootstrap.go
@@ -19,6 +19,7 @@ import (
 	"fmt"
 	"path/filepath"
 	"runtime"
+	"sort"
 	"strings"
 
 	"github.com/google/blueprint"
@@ -833,8 +834,15 @@ func (s *singleton) GenerateBuildActions(ctx blueprint.SingletonContext) {
 			pool = "console"
 		}
 
+		envKeys := make([]string, 0, len(i.Env))
+		for k := range i.Env {
+			envKeys = append(envKeys, k)
+		}
+		sort.Strings(envKeys)
+
 		envAssignments := ""
-		for k, v := range i.Env {
+		for _, k := range envKeys {
+			v := i.Env[k]
 			// NB: This is rife with quoting issues but we don't care because we trust
 			// soong_ui to not abuse this facility too much
 			envAssignments += k + "=" + v + " "
